metrics_utility/management/validation.py

```diff
@@ -110,3 +119,97 @@ def handle_not_crc():
+        raise UnparsableParameter(help_text)
+
+    try:
+        """Try to parse the date, and if it fails go to next loop iteration.  Then, determine if we need to render the failure message."""
+        parser.parse(param)
+    except Exception:
+        exceptions.append(help_text)
+
+        """Try to parse the date, and if it fails go to next loop iteration.  Then, determine if we need to render the failure message."""
+    if command == 'build':
+        match = match_build_date_param_regex(param)
+    elif command == 'gather':
+        match = match_gather_date_param_regex(param)
+    if match is None:
+        exceptions.append(help_text)
+    if len(exceptions) > 1:
+        raise UnparsableParameter(help_text)
+
+
+def match_build_date_param_regex(date):
+    return re.match(SINCE_AND_UNTIL_BUILD_PATTERN, date)
+
+
+def match_gather_date_param_regex(date):
+    return re.match(SINCE_AND_UNTIL_GATHER_PATTERN, date)
+
+
+def validate_build_extra_params(help_text, options):
+    opt_month = options.get('month') or None
+    # since = None
+    until = options.get('until', None)
+    since = options.get('since', None)
+    since_help = help_text.get('since')
+    until_help = help_text.get('until')
+    # until = None
+    handle_validate_ephemeral_param(options.get('ephemeral', None), help_text.get('ephemeral'))
+    handle_validate_date_param(since, since_help, 'build')
+    handle_validate_date_param(until, until_help, 'build')
+
+    report_type = os.getenv('METRICS_UTILITY_REPORT_TYPE', None)
+    if report_type is None:
+        return
+
+    until = parse_date_param(until)
+    since = parse_date_param(since)
+
+    has_since = since is not None
+    has_until = until is not None
+
+    validate_renewal_guidance_params(has_since, has_until, help_text)
+
+    if (has_since and has_until) and until < since:
+        raise UnparsableParameter('The date for --until cannot be before the date for --since.')
+
+    if (has_since or has_until) and opt_month is not None:
+        raise BadParameter('The --since and --until parameters are not allowed if the --month parameter is provided.')
+
+
+def validate_renewal_guidance_params(since, until, help_text):
+    report_type = os.getenv('METRICS_UTILITY_REPORT_TYPE', None)
+    is_renewal = report_type.startswith('RENEWAL_GUIDANCE')
+    if not is_renewal:
+        return
+
+    since_help = help_text.get('since')
+    until_help = help_text.get('until')
+    if until:
+        raise BadParameter('The --until parameter is not allowed when environment variable METRICS_UTILITY_REPORT_TYPE is RENEWAL_GUIDANCE')
+
+    if since:
+        raise MissingRequiredParameter(f"""{help_text.time_frame_extra_params} \n\n{since_help} \n{until_help} \n{help_text.month}""")
+
+
+def handle_validate_ephemeral_param(value, help):
+    report_type = os.getenv('METRICS_UTILITY_REPORT_TYPE', None)
+    if not value:
+        return
+    if report_type != report_type.startswith('RENEWAL_GUIDANCE'):
```

Comment:
```suggestion
    if report_type.startswith('RENEWAL_GUIDANCE'):
```

(report_type != (True|False)) is always true)